src/components/panel/panel.js

```diff
@@ -1737,14 +1741,36 @@ MdPanelRef.prototype._configureClickOutsideToClose = function() {
     // panel we don't want to panel to close.
     var self = this;
     var mouseupHandler = function(ev) {
-      if (sourceElem === target[0] && ev.target === target[0]) {
+      if (self.config['propagateContainerEvents']) {
+
+        // We check if the sourceElem of the event is the panel element or one
+        // of it's children. If it is not, then close the panel.
+        var found = findSourceElem(
+          self.panelEl.find(sourceElem.tagName),
+          sourceElem
+        );
+        if (!found && self.panelEl[0] !== sourceElem) {
+          self.close();
+        }
+
+      } else if (sourceElem === target[0] && ev.target === target[0]) {
         ev.stopPropagation();
         ev.preventDefault();
 
         self.close();
       }
     };
 
+    var findSourceElem = function(elems, sourceElem) {
+      var found = false;
+      angular.forEach(elems, function(elem) {
+        if (elem === sourceElem) {
+          found = true;
```

Comment:
How about this? It quits early when it finds a match and removes the extra variable.

```
var findSourceEl = function(els, sourceEl) {
  for (var el, i = 0; el = els[i]; i++) {
    if (el === sourceEl) {
      return true;
    }
  });
  return false;
};
```